libqtile/group.py

```diff
@@ -161,7 +188,7 @@ def layout_all(self, warp=False):
                         logger.exception("Exception in layout %s", self.layout.name)
                 if floating:
                     self.floating_layout.layout(floating, screen_rect)
-                if self.current_window and self.screen == self.qtile.current_screen:
+                if self.current_window and self.screen == self.qtile.current_screen and not FocusLock.is_active():
```

Comment:
This seems like it will break the if/else chain here and clear `last_focused` when we don't really want it to. Bud do we need this at all, given that you are overriding in focus() directly?

Reply:
Yes, it is necessary as `layout_all()` may be called without focus (e.g. below). I guess one could keep the `last_focused` record though.

Reply:
But if we're called without focus (here I assume you mean that `self.screen != self.qtile.current_screen`), won't we not call focus() here, so there's no focus to steal?

Reply:
It's called in line 242 with self.screen == self.qtile.screen.
And line 242 is necessary as otherwise there'll be no layouting and a window will be floating around at the top left corner, if the prompt widget has focus.

Reply:
> And line 242 is necessary as otherwise there'll be no layouting and a window will be floating around at the top left corner, if the prompt widget has focus while a new window is spawned.

Oh, I see, that makes it mutually recursive. Scary :(